staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -37,7 +38,22 @@ import (
 // TODO move to API machinery and re-unify with kubelet/server/portfoward
 const PortForwardProtocolV1Name = "portforward.k8s.io"
 
-var ErrLostConnectionToPod = errors.New("lost connection to pod")
+var (
+	// error returned whenever we lost connection to a pod
+	ErrLostConnectionToPod = errors.New("lost connection to pod")
+
+	// set of error we're expecting during port-forwarding
+	networkClosedError = "use of closed network connection"
+	// Containerd will return "network namespace for sandbox %q is closed", see:
+	// - 2.0: https://github.com/containerd/containerd/blob/v2.0.0-rc.6/internal/cri/server/sandbox_portforward_linux.go#L47
+	// - 1.7: https://github.com/containerd/containerd/blob/release/1.7/pkg/cri/server/sandbox_portforward_linux.go#L47
+	// CRI-O will return "sandbox %s is not running", see:
+	// - 1.31: https://github.com/cri-o/cri-o/blob/release-1.31/server/container_portforward.go#L54
```

Comment:
do we have pull requests open / acked to these locations to make sure they never change the format of that message, since these are now load-bearing in client-go?

are there other container runtimes (or modes within existing container runtimes) we have to worry about? docker? something on windows? gvisor? kata containers?

Reply:
this is this one https://github.com/golang/go/blob/76a8409eb81eda553363783dcdd9d6224368ae0e/src/internal/poll/fd.go#L24

Reply:
we should use https://github.com/golang/go/issues/4373 
`(errors.Is(err, net.ErrClosed)`

Reply:
> do we have pull requests open / acked to these locations to make sure they never change the format of that message, since these are now load-bearing in client-go?

I don't have any assurance like that, but when I checked the history of this piece of code it hasn't changed for several past releases. 

> are there other container runtimes (or modes within existing container runtimes) we have to worry about? docker? something on windows? gvisor? kata containers?

I'm definitely open to adding additional ones as we go, but crio and containerd being the two major players give us a good starting point.

Reply:
> this is this one https://github.com/golang/go/blob/76a8409eb81eda553363783dcdd9d6224368ae0e/src/internal/poll/fd.go#L24

> we should use [golang/go#4373](https://github.com/golang/go/issues/4373) `(errors.Is(err, net.ErrClosed)`

@aojea not quite sure what you're referring to?

Reply:
The error we are matching here is so common that is exposed by the standard library, and maybe using it instead our constant string it's safer , but that can be done later in any vaset

Reply:
Disused this offline with Antonio, his comments below were about that, which will allow us to drop one string matching error from `io.Copy` in favor of `errors.Is(err, net.ErrClosed)`